src/CMakeLists.txt

```diff
@@ -9,6 +9,6 @@ set(SOURCES
 set(CMAKE_EXE_LINKER_FLAGS "-shared -fPIC ${CMAKE_EXE_LINKER_FLAGS}")
 
 add_library(${PROJECT_NAME} SHARED ${SOURCES})
-target_link_libraries(${PROJECT_NAME} -ldl)
+target_link_libraries(${PROJECT_NAME} -ldl ${Boost_LIBRARIES})
```

Comment:
yes I know, I added it in case we will use any other Boost lib in future. i'll remove it, though